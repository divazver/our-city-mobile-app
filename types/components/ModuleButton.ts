export type ModuleButtonProps = {
    name: string,
    icon: string,
    link: string,
}