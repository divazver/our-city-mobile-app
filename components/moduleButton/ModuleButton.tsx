// @ts-nocheck
import {Colors, Spacings, View} from "react-native-ui-lib";
import {Link} from "expo-router";
import {Pressable, StyleSheet} from "react-native";
import AntDesign from "@expo/vector-icons/AntDesign";
import {ThemedText} from "@/components/ThemedText";
import {ModuleButtonProps} from "@/types/components/ModuleButton";

const ModuleButton = ({name, icon, link}: ModuleButtonProps) => {
    return <View style={styles.moduleWrapper}>
        <Link href={link} asChild style={styles.module}>
            <Pressable>
                <View style={styles.moduleIcon}><AntDesign name={icon} size={48}
                                                           color={Colors.$textDefaultLight}/></View>
                <View style={styles.moduleTitle}><ThemedText h6
                                                             style={styles.moduleTitleText}>{
                    name.length > 20 ? `${name.slice(0, 17)}...` : name
                }</ThemedText>
                </View>
            </Pressable>
        </Link>
    </View>;
}

const styles = StyleSheet.create({
    moduleWrapper: {
        width: '29.74%',
        display: "flex",
        alignItems: "center",
    },
    module: {
        width: "100%"
    },
    moduleIcon: {
        backgroundColor: Colors.$backgroundPrimaryHeavy,
        height: 105,
        borderRadius: 15,
        width: "100%",
        display: "flex",
        alignItems: "center",
        justifyContent: "center"
    },
    moduleTitle: {
        width: "100%",
        marginTop: Spacings.s1
    },
    moduleTitleText: {
        textAlign: "center",
    }
});

export default ModuleButton;