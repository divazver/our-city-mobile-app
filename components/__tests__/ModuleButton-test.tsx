import * as React from 'react';
import renderer from 'react-test-renderer';
import ModuleButton from "@/components/moduleButton/ModuleButton";

describe('<ModuleButton />', () => {

    // Snapshot test
    it('renders correctly', () => {
        const tree = renderer
            .create(<ModuleButton name={"Aktuality"} icon={"home"} link={"/"} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    // Unit test
    it('shorten module name', () => {
        const tree = renderer.create(<ModuleButton name={"Very long name of module!"} icon={"home"} link={"/"} />);

        expect(tree.root.findByProps({h6: true}).props.children.length).toBe(20);
    });
});

