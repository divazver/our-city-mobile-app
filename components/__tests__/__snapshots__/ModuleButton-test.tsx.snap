// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModuleButton /> renders correctly 1`] = `
<View
  style={
    [
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      {
        "alignItems": "center",
        "display": "flex",
        "width": "29.74%",
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    role="link"
    style={
      {
        "width": "100%",
      }
    }
  >
    <View
      style={
        [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "alignItems": "center",
            "backgroundColor": "#5A48F5",
            "borderRadius": 15,
            "display": "flex",
            "height": 105,
            "justifyContent": "center",
            "width": "100%",
          },
        ]
      }
    >
      <Text />
    </View>
    <View
      style={
        [
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "marginTop": 4,
            "width": "100%",
          },
        ]
      }
    >
      <Text
        h6={true}
        style={
          [
            {
              "backgroundColor": "transparent",
              "color": "#20303C",
              "writingDirection": "ltr",
            },
            undefined,
            undefined,
            undefined,
            undefined,
            {},
            undefined,
            undefined,
            undefined,
            [
              {
                "textAlign": "center",
              },
            ],
          ]
        }
      >
        Aktuality
      </Text>
    </View>
  </View>
</View>
`;
